### PR TITLE
feat: enhance null framework for standalone OAuth usage

### PR DESCRIPTION
**Pull request by brettchaldecott**
# Explain your changes

- Refactor OAuth initialization to use null framework for standalone usage
- Transform NullFramework into singleton pattern with session management
- Add thread-safe session handling with context variables support
- Update auth module exports for better framework integration
- Add comprehensive HTTP OAuth server example
- Include comprehensive test suite for null framework functionality

This enables standalone OAuth usage without requiring a web framework, making the SDK more flexible for various application architectures.

# Checklist

- [ ] I have read the [“Pull requests” section](https://github.com/kinde-oss/.github/blob/main/.github/CONTRIBUTING.md#pull-requests) in the [contributing guidelines](https://github.com/kinde-oss/.github/blob/main/.github/CONTRIBUTING.md).
- [ ] I agree to the terms within the [code of conduct](https://github.com/kinde-oss/.github/blob/main/.github/CODE_OF_CONDUCT.md).

🛟 _If you need help, consider asking for advice over in the [Kinde community](https://thekindecommunity.slack.com)._

### CODE REVIEW

File: kinde_sdk/auth/oauth.py [hunk 1]
```diff
@@ -365,8 +386,7 @@ async def login(self, login_options: Dict[str, Any] = None) -> str:
         Returns:
             Login URL
         """
-        if not self.framework:
-            raise KindeConfigurationException("Framework must be selected")
+        # Framework check removed - null framework is now supported
```
Comment (DanielRivers) [inline kinde_sdk/auth/oauth.py:389]: Please remove this comment

File: kinde_sdk/auth/oauth.py [hunk 2]
```diff
@@ -407,8 +427,7 @@ async def register(self, login_options: Dict[str, Any] = None) -> str:
         Returns:
             Registration URL
         """
-        if not self.framework:
-            raise KindeConfigurationException("Framework must be selected")
+        # Framework check removed - null framework is now supported
```
Comment (DanielRivers) [inline kinde_sdk/auth/oauth.py:430]: Please remove this comment

File: kinde_sdk/auth/oauth.py [hunk 3]
```diff
@@ -440,8 +459,7 @@ async def logout(self, user_id: Optional[str] = None, logout_options: Dict[str,
         Returns:
             Logout URL
         """
-        if not self.framework:
-            raise KindeConfigurationException("Framework must be selected")
+        # Framework check removed - null framework is now supported
```
Comment (DanielRivers) [inline kinde_sdk/auth/oauth.py:462]: Please remove this comment

File: kinde_sdk/auth/__init__.py [hunk 4]
```diff
@@ -4,9 +4,10 @@
...
 from .permissions import permissions
 from .claims import claims
+from .async_claims import async_claims
 from .feature_flags import feature_flags
 from .portals import portals
 from .tokens import tokens
 from .roles import roles
 
-__all__ = ["OAuth", "TokenManager", "UserSession", "permissions", "ApiOptions", "claims", "feature_flags", "portals", "tokens", "roles"]
+__all__ = ["OAuth", "TokenManager", "UserSession", "permissions", "ApiOptions", "claims", "async_claims", "feature_flags", "portals", "tokens", "roles"]
```
Comment (DanielRivers) [inline kinde_sdk/auth/__init__.py:13]: I don't see any reference to async_claims in the PR?

File: kinde_sdk/auth/oauth.py (hunk 2 above)
Comment (brettchaldecott) [reply kinde_sdk/auth/oauth.py]: I have removed this comment

File: kinde_sdk/auth/oauth.py (hunk 1 above)
Comment (brettchaldecott) [reply kinde_sdk/auth/oauth.py]: I have removed this comment
